WordPress/Classes/Services/BlogService.m

```diff
@@ -159,9 +159,14 @@ - (void)syncBlogsForAccount:(WPAccount *)account
 {
     DDLogMethod();
 
+    NSManagedObjectID *accountID = account.objectID;
     id<AccountServiceRemote> remote = [self remoteForAccount:account];
     [remote getBlogsWithSuccess:^(NSArray *blogs) {
         [self.managedObjectContext performBlock:^{
+            WPAccount *account = (WPAccount *)[self.managedObjectContext existingObjectWithID:accountID error:nil];
+            if (!account) {
+                return;
```

Comment:
> Always assume that the calls to services need to be done on the main context

Main/derived context has nothing to do with this issue. The problem here was that the caller just created/inserted the account object and hadn't saved the context yet, so the ID wasn't permanent and `existingObjectWithID` returned `nil`. It all happened on the main thread.